addons/product/models/product_pricelist.py

```diff
@@ -81,6 +81,13 @@ def write(self, values):
 
         return res
 
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        default = dict(default or {})
+        if 'name' not in default:
+            default['name'] = _("%s (copy)", self.name)
+        return super().copy(default=default)
```

Comment:
why don't you override `copy_data` instead ?

Also, please note that you'll have to adapt your override in 17.2 (see #154132)

Reply:
Thanks for the suggestion. overriding copy_data makes more sense. I have updated the code to do that and we will adapt the method as per new changes from 17.2.